src/ContextualPopup/ContextualPopup.js

```diff
@@ -538,19 +553,41 @@ module.exports = kind(
 	/**
 	* @private
 	*/
-	showHideScrim: function (inShow) {
-		if (this.floating && (this.scrim || (this.modal && this.scrimWhenModal))) {
-			var scrim = this.getScrim();
-			if (inShow && this.modal && this.scrimWhenModal) {
-				// move scrim to just under the popup to obscure rest of screen
-				var i = this.getScrimZIndex();
-				this._scrimZ = i;
-				scrim.showAtZIndex(i);
-			} else {
-				scrim.hideAtZIndex(this._scrimZ);
-			}
-			util.call(scrim, 'addRemoveClass', [this.scrimClassName, scrim.showing]);
+	updateScrim: function (old, value, source) {
+		if (this._updateScrimMutex) return;
+		this._updateScrimMutex = true;
+
+		// We sync modal and spotlightModal here because binding doesn't
+		// guarantee sequence when it is used with observers.
+		if (source == 'modal') this.set('spotlightModal', this.modal);
+		if (source == 'spotlightModal') this.set('modal', this.spotlightModal);
```

Comment:
We can have two way for setting spotlightModal value.
1) Set property on inherited kind.
- The modal value is copied from spotlightModal on initComponents. 
2) Set binding on inherited kind.
- The updateScrim is called from observer, but not copied from spotlightModal yet in this call.

So, below code is the related use case.

```
var myC = kind({
	name: 'myC',
	kind: ContextualPopup,
	bindings: [
		{from: 'bs', to: 'spotlightModal'}
	]
})
```